cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -436,6 +463,34 @@ private boolean isRunning(ProgramId programId) throws Exception {
     return ProgramStatus.STOPPED != getProgramStatus(programId);
   }
 
+  /**
+   * Returns the program status.
+   * @param programId the id of the program for which the running status in all versions of the app is found
+   * @return whether the same program is running in any versions of the app
+   * @throws NotFoundException if the application to which this program belongs was not found
+   */
+  private boolean isRunningInSameProgram(ProgramId programId) throws Exception {
+    // check that app exists
+    Collection<ApplicationId> appIds = store.getAllAppVersionsAppIds(programId.getNamespace(),
+                                                                     programId.getApplication());
+    if (appIds == null) {
+      throw new NotFoundException(Ids.namespace(programId.getNamespace()).app(programId.getApplication()).toId());
```

Comment:
don't use Ids class. Use new NamespaceId. Or in this cas, just new ApplicationId. Infact, you should use Id.Application here to avoid creating multiple objects, and also because NotFoundException behaves differently for ``Id`` and ``Object`` parameters